packages/cli/src/index.js

```diff
@@ -97,6 +106,78 @@ config({
   multiline: true,
 })
 
+const TELEMETRY_ENABLED =
+  telemetry !== 'false' && !process.env.REDWOOD_DISABLE_TELEMETRY
+
+async function main() {
+  // Start telemetry if it hasn't been disabled
+  if (TELEMETRY_ENABLED) {
+    try {
+      await startTelemetry()
+    } catch (error) {
+      console.error('Telemetry startup error')
+      console.error(error)
+    }
+  }
+
+  // Execute CLI within a span, this will be the root span
+  const tracer = trace.getTracer('redwoodjs')
+  await tracer.startActiveSpan('cli', async (span) => {
+    try {
+      // Run the command via yargs
+      await runYargs()
+
+      // Span housekeeping
+      span?.setStatus({ code: SpanStatusCode.OK })
+    } catch (error) {
+      const errorReferenceCode = uuidv4()
+      const errorMessage = [
+        error.stack ?? error.toString(),
+        '',
+        '-'.repeat(process.stdout.columns - 8),
+        '',
+        'Need help?',
+        ` - Not sure about something or need advice? Reach out on our ${terminalLink(
+          'Forum',
+          'https://community.redwoodjs.com/'
+        )}`,
+        ` - Think you've found a bug? Open an issue on our ${terminalLink(
+          'GitHub',
+          'https://github.com/redwoodjs/redwood'
+        )}`,
+        ` - Here's your unique error reference: '${errorReferenceCode}'`,
+      ].join('\n')
+      console.error(
+        boxen(errorMessage, {
+          padding: 1,
+          borderColor: 'red',
+          title: `Unexpected Error`,
+          titleAlignment: 'left',
+        })
+      )
+
+      recordTelemetryError(error)
+      recordTelemetryAttributes({ errorReferenceCode })
+      if (process.exitCode !== 0) {
+        process.exitCode ??= error.exitCode ?? 1
+      }
```

Comment:
I'd think the check for it not equaling zero wouldn't be necessary but maybe a command sets it to zero even though, is that the thinking? Or does it equal zero by default?

Reply:
The default value is undefined according to https://nodejs.org/api/process.html#processexitcode

I guess we do need to be careful here. We probably should remove the if condition and just always have a non-zero exit code on an error. If we don't want that then I guess we should fall back to just having the try/catch in the handler.

Reply:
Removing the conditions sounds like the way to go

Reply:
And `process.exitCode ??=` may not be enough either, since if a command sets it to process.exitCode = 1, it'll just pass it through. Not sure if that's a real concern since it's all our code, but may be more of a concern when there's plugins. That's in the future though

Reply:
I had been aiming for this past through behaviour. Thinking about the lint command. We likely don't want to throw an error when it exists non-zero - because it just always prints out the information to the output. However, we still want to maintain the same exit code so things like CI continue to work.